src/rust/engine/src/tasks.rs

```diff
@@ -167,6 +167,10 @@ impl Tasks {
         product: types.process_result,
         input: types.multi_platform_process_request,
       },
+      Intrinsic {
+        product: types.directory_digest,
```

Comment:
Thoughts on directly returning a `Snapshot`? `Snapshot`s are more ergonomic than directory digests and it fits well with the name `SnapshotSubset`. Conceptually, you are passing a `Snapshot` and getting back a (possibly) simpler `Snapshot`.

Reply:
@Eric-Arellano : A `Snapshot` can be lifted from a `Digest` (see line 146 in here), so `Get[Snapshot](SnapshotSubset(..))` should work.